Automattic/wp-calypso
# Gutenboarding: Update site launch styling.

#### Note to anyone reading this while preparing a release of the FSE plugin
This change doesn't require any release notes. We're developing a new feature which is currently hidden behind a flag.

## Changes proposed in this Pull Request

* **Standardized title, subtitle & action buttons.**
  Uses shared component from `@automattic/onboarding` package.

* **Fixed styling discrepancies between gutenboarding & launch.**
  More verbose styling is added to plans grid, domain picker, onboarding package to ensure they look the same under Calypso or FSE.

* **Added mobile support.** 
   Go back button turns into footer bar on mobile view. Container margins changes on different viewport width.

* **`<Modal>` component's header replaced with our own custom header.**
  This is for more flexibility like showing WP's logo on the header.

* **Domain categories is no longer shown as seen in the design**

## Testing instructions

**Getting started**
* Run `yarn` to rebuild all packages.
* Run `yarn dev --sync` on `app/full-site-editing` to sync to sandbox.
* Run `yarn start`.

**Setting up unlaunched site**
* Create an unlaunched site via `/new`.
* Once landed in the editor, point the hostname of the unlaunched site to your sandbox on your hosts file.
* Reopen browser and reopen your editor of your unlaunched site, e.g. `http://calypso.localhost:3000/block-editor/page/your_unlaunched_site.word**ess.com/home`.

**Test Launch Modal**
* Click on **Launch Site** button.
* Everything should look identical to how it looks like in Gutenboarding.
* Resize the browser viewport, side margins should change, **Go back** button on plans page should become a sticky footer.
* You should also see a WP logo on modal header.

**Test Gutenboarding**
* Just make sure there's no regression on `/new`.

## Screenshot
![2020-07-13_05-32-42 (1)](https://user-images.githubusercontent.com/1287077/87268384-83afdd00-c4ca-11ea-87b0-ffac2ac3bfda.gif)

Fixes part of #43750
